salt/_extensions/pillar/ca.py

```diff
@@ -320,7 +320,11 @@ def ext_pillar(minion_id, pillar, base="/etc/ssl", name="PSFCA", cert_opts=None)
     gen_certs = pillar.get("tls", {}).get("gen_certs", {})
     for certificate, config in gen_certs.items():
         role_patterns = [
-            pillar.get("roles", {}).get(r) for r in config.get("roles", "")
+            role.get("pattern")
```

Comment:
This change is necessary due to the new structure of the pillar key.

Note that we need to filter out the `None` values that would result from values that do not define a `pattern` key in their contents... which would ultimately not work anyway. That is why we nest the two list comprehensions and add the if clause.